src/components/Modal/VaultStrategyModal/managedVaults/ManagedVaultsStrategyModalAllocations.tsx

```diff
@@ -0,0 +1,181 @@
+                    </div>
+                  </div>
+                  <div className="flex h-[24px] w-full flex-none flex-row items-start gap-1 self-stretch p-0">
+                    <div className="w-1/2 items-center text-sm font-semibold capitalize text-pink-50">
+                      Ratio :
+                    </div>
+                    <div className="w-1/2 items-center text-right text-sm">
+                      {formatPercentage(
+                        (data.datasets?.[0]?.data?.[i] * 100) / sumTotal / 100
+                      )}
+                    </div>
+                  </div>
+                </div>
+              </div>
+            </div>
+          ))}
+        </div>
+      </div>
+      <div className="flex w-full flex-none grow-0 flex-col items-start gap-2 p-0">
+        {data.labels?.map((label, i) => (
+          <div
+            key={`legend-${i}`}
+            className="flex h-[24px] w-full flex-none grow-0 flex-row items-center gap-2 self-stretch p-0"
+          >
+            <div
+              className="h-3 w-3 flex-none grow-0 border"
+              style={{
+                borderColor: `${data.datasets?.[0].borderColor[i]}`,
+                backgroundColor: `${data.datasets?.[0].backgroundColor[i]}`,
+              }}
+            ></div>
+            <div className="flex w-[calc(50%-30px)] flex-none grow-0 flex-row items-center gap-1 p-0">
+              <div className="relative float-left mr-1 h-6 w-6 rounded-full bg-white">
+                <AssetLogo tokenAddress={addresses[i]} />
+              </div>
+              <span className="text-left text-xs uppercase text-pink-50">
+                {label}
+              </span>
+            </div>
+            <div className="grow-1 flex w-1/2 flex-none flex-row items-center gap-1 p-0">
+              <div className="w-full text-right text-sm text-pink-50">
+                {formatPercentage(
+                  (data.datasets?.[0]?.data?.[i] * 100) / sumTotal / 100
+                )}
+              </div>
+            </div>
+          </div>
+        ))}
+      </div>
+    </div>
+  )
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const externalTooltipHandler = (context: any) => {
+  // Tooltip Element
+  const { chart, tooltip } = context
+  const elements = document.querySelectorAll("[data-tooltip]")
+  elements.forEach((element) => {
+    const htmlElement = element as HTMLElement
+    htmlElement.style.display = "none"
+  })
+
+  if (tooltip.opacity !== 0) {
+    const titleLines = tooltip.title || []
+    const tooltipElts = titleLines.map((title: string) => {
+      return chart.canvas.parentNode.querySelector(`[data-tooltip="${title}"]`)
+    })
+
+    const { offsetLeft: positionX, offsetTop: positionY } = chart.canvas
+
+    const tooltipElt = tooltipElts?.[0]
+    if (!tooltipElt) {
+      return
+    }
+    // Display, position, and set styles for font
+    tooltipElt.style.pointerEvents = "none"
```

Comment:
Not super react-y stuff here but that is sometimes the nature of these libraries.

Reply:
yes, I don't have much choice here. Even the official documentation recommends doing this way.

Reply:
Yeah, I understand.